src/qofonomanager.cpp

```diff
@@ -106,6 +106,7 @@ void QOfonoManager::onModemRemoved(const QDBusObjectPath& path)
 void QOfonoManager::onGetModemsFinished(QDBusPendingCallWatcher* watcher)
 {
     QDBusPendingReply<ObjectPathPropertiesList> reply(*watcher);
+    watcher->deleteLater();
```

Comment:
Even then, deferred deletes don't get processed until the outer event loop is reached.